source/boundary_temperature/function.cc

```diff
@@ -107,22 +148,31 @@ namespace aspect
       prm.enter_subsection("Boundary temperature model");
       {
         prm.enter_subsection("Function");
-        try
-          {
-            boundary_temperature_function.parse_parameters (prm);
-          }
-        catch (...)
-          {
-            std::cerr << "ERROR: FunctionParser failed to parse\n"
-                      << "\t'Boundary temperature model.Function'\n"
-                      << "with expression\n"
-                      << "\t'" << prm.get("Function expression") << "'"
-                      << "More information about the cause of the parse error \n"
-                      << "is shown below.\n";
-            throw;
-          }
-        min_temperature = prm.get_double ("Minimal temperature");
-        max_temperature = prm.get_double ("Maximal temperature");
+        {
+          if (prm.get ("Coordinate system") == "cartesian")
```

Comment:
can you move this into a helper function ``Utilities::Coordinates::string_to_coordinate_system(string)``